### ISSUE TITLE
Enhancement: name lib and grammar generator with name of the grammar type

### ISSUE THREAD

**Issue by vanhauser-thc** [AFLplusplus/Grammar-Mutator#12]
when compiling for a grammar compiling new would overwrite an existing one. also from the lib name it is not clear for which grammar it is.

proposal: simply name the resulting grammar generator and lib with the grammar type, e.g. `libgrammarmutator-json.so`

**Comment by h1994st** [comment]
I was thinking about this.

What about appending the grammar file name to the generated library? It looks a bit long. For example:
- Grammar file: `json_grammar.json`
- Generated library: `libgrammarmutator_json_grammar.so`

Or let the user specify a short name, so the name of the generated library can be shorter. Otherwise, we can fall back to the default naming, by appending the grammar file name to the library name.

**Comment by h1994st** [comment]
Updated the building scripts to append a suffix to the grammar mutator lib and `grammar_generator`.

Please take a look at the latest commits in `dev` branch

**Comment by vanhauser-thc** [comment]
I made a few changes, e.g. renaming to lowercase letter and removing everything after _.- to keep the name short.
also made it a make option.

additionally I updated the documentation.

a few other things that could be done:
  * rename all files to lowercase
  * the RNG parameter for the grammar-generator could be made option and moved to the last position (and then e.g. time() being used as seed)

but I leave that decision to you.
